packages/shared/src/lib/query.ts

```diff
@@ -10,11 +10,22 @@ import { EmptyObjectLiteral } from './kratos';
 import { LoggedUser } from './user';
 import { FeedData, Post, ReadHistoryPost } from '../graphql/posts';
 import { ReadHistoryInfiniteData } from '../hooks/useInfiniteReadingHistory';
+import { MainFeedPage } from '../components/utilities';
+
+export enum OtherFeedPage {
+  Tag = 'tag',
+  Squad = 'squad',
+  Source = 'source',
+  Bookmarks = 'bookmarks',
+  SearchBookmarks = 'search-bookmarks',
+}
+
+export type AllFeedPages = MainFeedPage | OtherFeedPage;
 
 export type MutateFunc<T> = (variables: T) => Promise<(() => void) | undefined>;
 
 export const generateQueryKey = (
-  name: string | RequestKey,
+  name: RequestKey | AllFeedPages,
```

Comment:
Can't we make the RequestKey extend AllFeedPages or Include them in some way there?
I don't want to start adding exceptions only in certain places.

Reply:
I wanted to do that too, but I found out enum is not extend-able 😭

However, I think this is the "only exception" for this implementation. So I kinda made my peace with it 😅

Reply:
I do think there are [ways](https://dev.to/afrazkhan/merge-typescript-enums-4mh7) to do it.
But it's not urgent so up to you :)

Reply:
Oh, nice. Thank you for the reference. I learned something new today. I think we can take this on after refactoring our feed flow. The querying part might be affected by the upcoming refactoring, so for the moment, I would try to keep the changes lean for now. Wdyt?

Reply:
Yeah no problem at all :)